### PR TITLE
add: qr scanner and generator

### PR DESCRIPTION
**Pull request by singodiyashubham87**
# Title and Issue number 

Title : QR Scanner and Generator

Issue No. : #249 

Code Stack : HTML, CSS, JS

Close #<issue_no>

# Checklist:

- [x] I have mentioned the issue number in my Pull Request.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have updated the Index.html file for my contribution

### CODE REVIEW

File: native/machine-coding/qr-scanner-and-generator/images/bg.jpg
Comment (sadanandpai) [inline native/machine-coding/qr-scanner-and-generator/images/bg.jpg:1]: no bg images are required

File: native/machine-coding/qr-scanner-and-generator/index.html [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/css/bootstrap.min.css"
+      rel="stylesheet"
+      integrity="sha384-9ndCyUaIbzAi2FUVXJi0CjmCapSmO7SnpJef0486qhLnuZ2cdeRhO02iuK6FUUVM"
+      crossorigin="anonymous"
+    />
+    <script
+      src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/js/bootstrap.bundle.min.js"
+      integrity="sha384-geWF76RCwLtnZ8qwWowPQNguL3RmwHVBC9FhGdlKrxdiJJigb/j/68SIy3Te4Bkz"
+      crossorigin="anonymous"
+    ></script>
```
Comment (sadanandpai) [inline native/machine-coding/qr-scanner-and-generator/index.html:22]: bootstrap is not needed

File: native/machine-coding/qr-scanner-and-generator/scripts/html5-qrcode.min.js
Comment (sadanandpai) [inline native/machine-coding/qr-scanner-and-generator/scripts/html5-qrcode.min.js:1]: lets import from cdn and not add it in codebase

File: native/machine-coding/qr-scanner-and-generator/scripts/qrcode.js [hunk 2]
```diff
@@ -0,0 +1,2297 @@
...
+
+(function (factory) {
+  if (typeof define === 'function' && define.amd) {
+      define([], factory);
+  } else if (typeof exports === 'object') {
+      module.exports = factory();
+  }
+}(function () {
+    return qrcode;
+}));
```
Comment (sadanandpai) [inline native/machine-coding/qr-scanner-and-generator/scripts/qrcode.js:2297]: use the modern standards. Also, divide the file into multiple reusable chunks

File: native/machine-coding/qr-scanner-and-generator/styles/styles.css [hunk 3]
```diff
@@ -0,0 +1,374 @@
+@import url("https://fonts.googleapis.com/css2?family=Ysabeau+Infant&display=swap");
```
Comment (sadanandpai) [inline native/machine-coding/qr-scanner-and-generator/styles/styles.css:1]: no new font needed

File: native/machine-coding/qr-scanner-and-generator/styles/styles.css [hunk 4]
```diff
@@ -0,0 +1,374 @@
+@import url("https://fonts.googleapis.com/css2?family=Ysabeau+Infant&display=swap"); 
+
+*{
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+  font-family: "Ysabeau Infant", sans-serif;
+}
```
Comment (sadanandpai) [inline native/machine-coding/qr-scanner-and-generator/styles/styles.css:8]: above css rules are not needed

**Review by sadanandpai — CHANGES_REQUESTED** [review]

**Comment by singodiyashubham87** [comment]
@sadanandpai Have a look now.

**Review by sadanandpai — CHANGES_REQUESTED** [review]

**Comment by sadanandpai** [comment]
Please address all the comments

**Comment by singodiyashubham87** [comment]
@sadanandpai Done!

File: native/machine-coding/qr-scanner-and-generator/scripts/html5-qrcode.min.js
Comment (sadanandpai) [inline native/machine-coding/qr-scanner-and-generator/scripts/html5-qrcode.min.js:1]: external library file should not be copied to the repo. Please import if cdn link is present

**Review by sadanandpai — CHANGES_REQUESTED** [review]

**Pull request closed without merging**